processing/src/main/java/io/druid/segment/IndexIO.java

```diff
@@ -91,12 +93,13 @@
   private static final SerializerUtils serializerUtils = new SerializerUtils();
 
   private final ObjectMapper mapper;
+  private final ColumnConfig columnConfig;
```

Comment:
The intention is the opposite, to move column config out of IndexIO: #1078, not to establish it as a part of IndexIO. And actually this PR is a good moment to move ColumnConfig at least one step down, to specific "SegmentStrategies" (those that need it).